src/VisualStudio/Core/Def/Implementation/TaskList/CommentTaskTokenSerializerProvider.cs

```diff
@@ -39,14 +40,10 @@ internal sealed class CommentTaskTokenSerializerProvider : IOptionPersisterProvi
         public async ValueTask<IOptionPersister> GetOrCreatePersisterAsync(CancellationToken cancellationToken)
         {
             if (_lazyPersister is not null)
-            {
                 return _lazyPersister;
-            }
-
-            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
             // Not all SVsTaskList implementations implement ITaskList, but when it does we will use it
-            var taskList = await _serviceProvider.GetServiceAsync(typeof(SVsTaskList)).ConfigureAwait(true) as ITaskList;
+            var taskList = await _serviceProvider.GetServiceAsync<SVsTaskList, ITaskList>(throwOnFailure: false).ConfigureAwait(false);
             _lazyPersister ??= new CommentTaskTokenSerializer(_optionService, taskList);
```

Comment:
Is `CommentTaskTokenSerializer` safe to initialize from a background thread?

Reply:
it is, yup.